djangocms_navigation/admin.py

```diff
@@ -128,6 +136,13 @@ def change_view(
         extra_context = extra_context or {}
         if menu_content_id:
             request.menu_content_id = menu_content_id
+            if self._versioning_enabled:
+                menu_content = get_object_or_404(
+                    MenuContent._base_manager, id=menu_content_id)
+                version = Version.objects.get_for_content(menu_content)
+                if version.state != DRAFT:
```

Comment:
What do you think about the following:
```
from djangocms_versioning.conditions import ConditionFailed
try:
    version.check_modify()
except ConditionFailed as e:
    messages.error(request, e)
```

This takes care of checking other conditions as they appear, for example moderation checks if the object is already in a collection to fail the check.